client.go

```diff
@@ -0,0 +1,97 @@
+package skynet
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"gitlab.com/NebulousLabs/errors"
+)
+
+type (
+	// SkynetClient is the Skynet Client which can be used to access Skynet.
+	SkynetClient struct {
+		PortalURL string
+		Options   Options
+	}
+
+	// requestOptions contains the options for a request.
+	requestOptions struct {
+		Options
+
+		method    string
+		reqBody   io.Reader
+		extraPath string
+		query     url.Values
+	}
+)
+
+// NewSkynetClient creates a new Skynet Client which can be used to access Skynet.
+// Pass in "" for the portal to let the function select one for you.
+func NewSkynetClient(portalURL string) SkynetClient {
+	if portalURL == "" {
+		portalURL = DefaultPortalURL
+	}
+	return SkynetClient{
+		PortalURL: portalURL,
+	}
+}
+
+// SetCustomOptions sets the custom options for this client.
+func (sc *SkynetClient) SetCustomOptions(customOptions Options) {
```

Comment:
I think it would be cleaner to have a `New()` and a `NewCustom` similar to how we do it within Sia.

Then the majority of the time devs can just call `skynet.NewClient()` and not pass in the empty string all the time. 

`NewClient()` in turn can call `NewCustom`

```
func NewClient() SkynetClient {
 return NewCustomClient("",Options)
}
```

Then devs just have the 1 call, either `New` or `NewCustom` vs needing two calls if then want custom options.

Reply:
The reason for this design was mainly compatibility with the other SDKs so the functions mostly looked the same. However in this case I think it would be better to be idiomatic and handle the differences in initialization on the documentation level. Thanks for the feedback 👍